commitlint.config.ts

```diff
@@ -57,11 +60,35 @@ module.exports = {
         // * Workflow: detect if wip commit in a branch not named "wip/*" or whose name contains "squashed".
         // * Allow PascalCase word after colon in title (exception to subject-lowercase rule), e.g.: "End2End: TestFixtureSetup refactor"
         // * Detect if commit hash mention in commit msg actually exists in repo.
-        // * Give replacement suggestions in rule that detects too long titles (e.g. and->&, config->cfg, ...)
         // * Detect area(sub-area) in the title that doesn't include area part (e.g., writing (bar) instead of foo(bar))
 
         {
             rules: {
+                'header-max-length-with-suggestions': ({header}: {header:any}) => {
+                    let headerStr = convertAnyToString(header, "header");
+                    let offence = false;
+
+                    let message = `Please do not exceed ${headerMaxLineLength} characters in title.`;
+                    if (headerStr.length > 50) {
+                        offence = true;
+                        let numRecomendations = 0;
+                        Object.entries(abbr).forEach(([key, value]) => {  
+                            if (headerStr.includes(key.toString())){
+                                if (numRecomendations === 0) {
+                                    message = message + 'We recommend the following replacement(s) in your commit title:\n'
```

Comment:
It's better to not use "we/I" in messages so let's change this to "The following replacement(s) in your commit title are recommended:"

Reply:
Done